### PR TITLE
CTC-48: add phonic outcomes to lesson overview page header

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 1997

- Renders phonic outcomes on the lesson overview header, updating relevant schema 
- Replaces deprecated flex and box components with comp lib versions

## Issue(s)

Fixes #CTC-48

### Acceptance Criteria:

- [ ]  Phonics outcome to be displayed on lesson overview page below the normal lesson outcome
- [ ]  Character limit of 190

## How to test

1. Go to https://deploy-preview-2965--oak-web-application.netlify.thenational.academy/teachers/beta/lessons/des-auteurs-francophones-perfect-tense-with-etre
2. You should see phonic outcomes on the mfl headers 

## Screenshots

How it should now look:

Screenshot 2024-11-06 at 16 46 35

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

**Review by RBrownsett — CHANGES_REQUESTED** [review]
@k-huggs I'm getting a 500 error for every lesson I try and open. An ideas? e.g . https://deploy-preview-2965--oak-web-application.netlify.thenational.academy/teachers/programmes/french-primary-ks2-l/units/ou-habites-tu-cade/lessons/saying-your-nationality-crup6d


image

**Comment by k-huggs** [comment]
> @k-huggs I'm getting a 500 error for every lesson I try and open. An ideas? e.g . [deploy-preview-2965--oak-web-application.netlify.thenational.academy/teachers/programmes/french-primary-ks2-l/units/ou-habites-tu-cade/lessons/saying-your-nationality-crup6d](https://deploy-preview-2965--oak-web-application.netlify.thenational.academy/teachers/programmes/french-primary-ks2-l/units/ou-habites-tu-cade/lessons/saying-your-nationality-crup6d)
> 
> image

Just looking into it now

**Review by Monika-Koziol — APPROVED** [review]

**Comment by k-huggs** [comment]
> @k-huggs I can see two outcomes - the one thing I have noticed that lessons with one outcome when viewed in mobile view the heading 'Learning Outcomes' is displayed. Is that expected?
> 
> Mobile: image
> 
> Desktop: image

Removed learning outcomes from mobile view

**Review by RBrownsett — APPROVED** [review]
Tested in mobile view and desktop against different browsers, second outcome is displayed.

@k-huggs in terms of regression no major issues, spotted in the Percy tests that the subject icons looks slightly smaller - but on the front end I can see no noticeable difference

File: src/node-lib/curriculum-api-2023/queries/lessonOverview/lessonOverview.query.ts [hunk 1]
```diff
@@ -265,11 +266,14 @@ const lessonOverviewQuery =
     const pathways = canonicalLesson ? getPathways(res) : [];
 
     lessonBrowseDataByKsSchema.parse(browseDataSnake);
-    lessonContentSchema.parse(contentSnake);
+    lessonContentSchema.parse({ ...contentSnake, phonics_outcome: null });
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/queries/lessonOverview/lessonOverview.query.ts:269]: why are we setting it to null?

Comment (k-huggs) [reply src/node-lib/curriculum-api-2023/queries/lessonOverview/lessonOverview.query.ts:269]: Phonic data is only on the new lesson content mv and the current lesson overview shares types with beta overview page

Comment (njblore) [reply src/node-lib/curriculum-api-2023/queries/lessonOverview/lessonOverview.query.ts:269]: gotcha thanks, do we have that documented anywhere atm?

**Review by njblore — APPROVED** [review]

File: src/node-lib/curriculum-api-2023/queries/lessonOverview/lessonOverview.query.ts (hunk 1 above)
Comment (k-huggs) [reply src/node-lib/curriculum-api-2023/queries/lessonOverview/lessonOverview.query.ts:269]: I'll drop a comment on it, there's gonna be more of this until some of these things become published i think

**Pull request merged**